controllers/rsyncnodedatamovement_controller.go

```diff
@@ -95,6 +105,34 @@ func (r *RsyncNodeDataMovementReconciler) Reconcile(ctx context.Context, req ctr
 		}
 	}
 
+	// Perform the data movement cancellation, if requested
+	if rsyncNode.Spec.Cancel {
```

Comment:
If rsync request is created but kubernetes sits on the object for a bit, then shortly after a cancel is issued, I think we could get a scenario where there is nothing to cancel. So you might need to add a check here
```
if rsyncNode.Status.StartTime.IsZero() {
  rsyncNode.Status.State|Staus|EndTime = Finished|Canceled|Now
  
  if err := r.Status().Update(ctx, rsyncNode); err != nil {
    return ctrl.Result{}, err
  }

  return ctrl.Result{}, nil
}

Reply:
Good catch.